src/region.js

```diff
@@ -204,12 +208,38 @@ Marionette.Region = Marionette.Object.extend({
     return Backbone.$(el, Marionette._getValue(this.options.parentEl, this));
   },
 
+  // Replace the region's DOM element with the view's DOM element.
+  replaceEl: function(view) {
+    // empty el so we don't save any non-destroyed views
+    this.$el.contents().detach();
```

Comment:
After talking to @ahumphreys87 in Gitter I think we're fine here. `preventDestroy` is a tool specifically for creating memory leaks so I think it's all good. ;-)